backend/src/bundles/achievements/achievement.controller.ts

```diff
@@ -0,0 +1,244 @@
+     *              schema:
+     *                 type: object
+     *                 $ref: '#/components/schemas/Achievements'
+     *        401:
+     *          description: Failed operation
+     *          content:
+     *              application/json:
+     *                  schema:
+     *                      type: object
+     *                      $ref: '#/components/schemas/Error'
+     */
+
+    private async findById(
+        options: ApiHandlerOptions<{
+            params: UserGetAllItemResponseDto;
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        try {
+            const { params } = options;
+            const achievement = await this.achievementService.findById(
+                params.id,
+            );
+
+            return {
+                type: ApiHandlerResponseType.DATA,
+                status: HttpCode.OK,
+                payload: achievement,
+            };
+        } catch {
+            return {
+                type: ApiHandlerResponseType.DATA,
+                status: HttpCode.NOT_FOUND,
+                payload: {
+                    errorType: 'COMMON',
+                    message: 'Achievement not found',
+                },
+            };
+        }
+    }
+
+    /**
+     * @swagger
+     * /api/v1/achievements/user/{id}:
+     *    get:
+     *      tags:
+     *       - UserAchievements
+     *      description: Returns achievements for a specific user
+     *      security:
+     *        - bearerAuth: []
+     *      parameters:
+     *        - name: id
+     *          in: path
+     *          description: ID of the user
+     *          required: true
+     *          schema:
+     *            type: number
+     *
+     *      responses:
+     *        200:
+     *          description: Successful operation
+     *          content:
+     *            application/json:
+     *              schema:
+     *                 type: object
+     *                 $ref: '#/components/schemas/Achievements'
+     *
+     *        401:
+     *          description: Failed operation
+     *          content:
+     *              application/json:
+     *                  schema:
+     *                      type: object
+     *                      $ref: '#/components/schemas/Error'
+     */
+
+    private async findByUserId(id: number): Promise<ApiHandlerResponse> {
+        try {
+            const userAchievements =
+                await this.achievementService.findByUserId(id);
+
+            return {
+                type: ApiHandlerResponseType.DATA,
+                status: HttpCode.OK,
+                payload: userAchievements,
+            };
+        } catch {
+            return {
+                type: ApiHandlerResponseType.DATA,
+                status: HttpCode.NOT_FOUND,
+                payload: {
+                    errorType: 'COMMON',
+                    message: 'User achievements not found',
+                },
+            };
+        }
+    }
```

Comment:
Is it necessary because i think you can just handle it in service and just throw an error if there are no achievements

Reply:
Thanks, I fixed it.